Modify S1104: Migrate To LayC - avoid public field (#3238)
SonarSource/rspec: rules/S1104/impacts.adoc

=== What is the potential impact?

> When any part of the code can modify a public field, this can lead to unexpected changes and hard-to-trace bugs.

Reviewer: What about rephrasing it?

```suggestion
Public fields can be modified by any part of the code and this can lead to unexpected changes and hard-to-trace bugs.
```

> When a field is public, it is no longer possible to change how the data is stored internally without impacting the client code of the class.

Author: It sounds much better, thanks 🤩
